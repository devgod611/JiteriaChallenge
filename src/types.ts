export type UserListType = {
  users: UserType[];
};

export type UserType = {
  email: string,
  id: number,
  name: string,
  phone: string,
  username: string,
  website: string,
};

export type UserState = {
  users: UserType[];
};

export type RootState = {
  UserState: UserState;
};
