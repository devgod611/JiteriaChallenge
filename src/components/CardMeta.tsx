import React, { ReactElement } from "react";
import { Card, Typography } from "antd";
import { MailOutlined, PhoneOutlined, GlobalOutlined } from "@ant-design/icons";

const { Meta } = Card;

function CardMeta({ item }: any): ReactElement {
  return (
    <>
      <Meta title={item.name} className="meta" />
      <Typography.Text className="card_text">
        <MailOutlined /> &nbsp; {item.email}
      </Typography.Text>
      <Typography.Text className="card_text">
        <PhoneOutlined /> &nbsp; {item.phone}
      </Typography.Text>
      <Typography.Text className="card_text">
        <GlobalOutlined /> &nbsp; {item.website}
      </Typography.Text>
    </>
  );
}

export default CardMeta;
