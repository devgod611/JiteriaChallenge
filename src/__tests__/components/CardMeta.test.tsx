import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import CardMeta from "../../components/CardMeta";
import { UserType } from "../../types";

describe("CardMeta", () => {
  describe("rendering", () => {
    let wrapper: ShallowWrapper;
    const item: UserType = {
        email: "Sincere@april.biz",
        id: 1,
        name: "Leanne Graham",
        phone: "1-770-736-8031 x56442",
        username: "Bret",
        website: "hildegard.org"
    };
    beforeEach(() => {
      wrapper = shallow(<CardMeta item={item} />);
    });

    it("renders the component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
