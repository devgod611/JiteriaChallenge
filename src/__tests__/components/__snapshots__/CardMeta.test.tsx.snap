// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMeta rendering renders the component 1`] = `
<Fragment>
  <Meta
    className="meta"
    title="Leanne Graham"
  />
  <Text
    className="card_text"
  >
    <ForwardRef(MailOutlined) />
       
    Sincere@april.biz
  </Text>
  <Text
    className="card_text"
  >
    <ForwardRef(PhoneOutlined) />
       
    1-770-736-8031 x56442
  </Text>
  <Text
    className="card_text"
  >
    <ForwardRef(GlobalOutlined) />
       
    hildegard.org
  </Text>
</Fragment>
`;
